ipa-core/src/hpke/registry.rs

```diff
@@ -62,6 +62,19 @@ pub trait PublicKeyRegistry {
     fn public_key(&self, key_id: KeyIdentifier) -> Option<&IpaPublicKey>;
 }
 
+// Similar as above with PublicKeyOnly, we need this type for when only the private key is required.
+pub struct PrivateKeyOnly {
+    sk: IpaPrivateKey,
+}
+
+impl From<IpaPrivateKey> for PrivateKeyOnly {
+    fn from(value: IpaPrivateKey) -> Self {
+        Self {
+            sk: value
+        }
+    }
+}
+
```

Comment:
Without this, I get this compile error:
```
error[E0277]: the trait bound `PrivateKeyOnly: From<hpke::dhkex::x25519::PrivateKey>` is not satisfied
   --> ipa-core/src/config.rs:245:31
    |
245 |     Ok(KeyRegistry::from_keys([IpaPrivateKey::from_bytes(&sk)?]))
    |        ---------------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<hpke::dhkex::x25519::PrivateKey>` is not implemented for `PrivateKeyOnly`, which is required by `hpke::dhkex::x25519::PrivateKey: Into<_>`
    |        |
    |        required by a bound introduced by this call
    |
    = note: required for `hpke::dhkex::x25519::PrivateKey` to implement `Into<PrivateKeyOnly>`
note: required by a bound in `KeyRegistry::<K>::from_keys`
   --> ipa-core/src/hpke/registry.rs:91:41
    |
91  |     pub fn from_keys<const N: usize, I: Into<K>>(pairs: [I; N]) -> Self {
    |                                         ^^^^^^^ required by this bound in `KeyRegistry::<K>::from_keys`
```

with it, I get:
```
error[E0119]: conflicting implementations of trait `From<PrivateKeyOnly>` for type `PrivateKeyOnly`
  --> ipa-core/src/hpke/registry.rs:70:1
   |
70 | impl From<IpaPrivateKey> for PrivateKeyOnly {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T> From<T> for T;
```

Reply:
I think it may be the same issue as @andyleiserson faced when implementing `PublicKeyOnly` (see the comment above) 

I am not sure why you need a wrapper around secret key - can you use `KeyRegistry<IpaPrivateKey>` instead?

Reply:
That's actually what I tried first, but got some other error, and went down this path (following what Andy had done.) Let me give that a try again.